### PR TITLE
Replaced JSONField with django.contrib.fields.JSONField. Updated test…

### PR DESCRIPTION
**Pull request by Killsoft1692**
…s. Removed django-jsonfield from requirements. Updated test_app.settings

### CODE REVIEW

File: audit_history/tests/test_audit_history.py [hunk 1]
```diff
@@ -16,25 +16,25 @@ def test_empty_history_after_save(self):
...
         self.assertEqual(None, self.blog_post.history[0]['actor'])
-        self.assertEqual(self.payload[1], self.blog_post.history[0]['payload'][1])
-        self.assertEqual(self.payload[2], self.blog_post.history[0]['payload'][2])
+        self.assertEqual(self.payload['a'], self.blog_post.history[0]['payload']['a'])
+        self.assertEqual(self.payload['b'], self.blog_post.history[0]['payload']['b'])
 
     def test_n_usages_of_save_with_audit(self):
         count = randint(1, 100)
-        for _ in xrange(count):
+        for _ in range(count):
```
Comment (hheimbuerger) [inline audit_history/tests/test_audit_history.py]: Just out of curiosity: why this change?

File: audit_history/tests/test_audit_history.py [hunk 2]
```diff
@@ -16,25 +16,25 @@ def test_empty_history_after_save(self):
...
     def test_save_with_audit_record_without_user(self):
         self.blog_post.save_with_audit_record(None, self.history_event, payload=self.payload)
         self.assertEqual(None, self.blog_post.history[0]['actor'])
-        self.assertEqual(self.payload[1], self.blog_post.history[0]['payload'][1])
-        self.assertEqual(self.payload[2], self.blog_post.history[0]['payload'][2])
+        self.assertEqual(self.payload['a'], self.blog_post.history[0]['payload']['a'])
+        self.assertEqual(self.payload['b'], self.blog_post.history[0]['payload']['b'])
 
     def test_n_usages_of_save_with_audit(self):
         count = randint(1, 100)
```
Comment (hheimbuerger) [inline audit_history/tests/test_audit_history.py]: Wouldn't it be better if this test was more deterministic and always test e.g. 100 cycles? What's the reason for the random cycle count? (Just trying to understand, I'm probably missing something here.)

File: audit_history/tests/test_audit_history.py (hunk 1 above)
Comment (Killsoft1692) [reply audit_history/tests/test_audit_history.py]: Because in python3 range == xrange in python2
https://github.com/espressopp/espressopp/issues/111

File: audit_history/tests/test_audit_history.py (hunk 2 above)
Comment (Killsoft1692) [reply audit_history/tests/test_audit_history.py]: The main idea is check that count of saving == count of history items(Save must works once as good as 100 times).

File: audit_history/mixins.py [hunk 3]
```diff
@@ -38,6 +42,7 @@ def _create_history_entry(modification_time, current_user, event, **payload):
                      'name': current_user.get_full_name(),
                      'is_staff': current_user.is_staff
                  }}
+        payload = json.loads(json.dumps(payload, default=json_formatter))
```
Comment (hheimbuerger) [inline audit_history/mixins.py:45]: I'm not sure I fully understand this. It's dumping it and then immediately loading it again? Basically converting datetimes into strings, for example?
Is there maybe a cleaner way to express this?

File: audit_history/utils.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import datetime
+from decimal import Decimal
+
+
+def json_formatter(o):
```
Comment (hheimbuerger) [inline audit_history/utils.py:5]: A lot of these are very one-directional, though, correct? As in, a datetime is rendered into a string, but never turned back into a datetime?

**Review by hheimbuerger — APPROVED** [review]

File: audit_history/tests/test_audit_history.py (hunk 2 above)
Comment (hheimbuerger) [reply audit_history/tests/test_audit_history.py]: But why randomize and not fix to e.g. 100?

File: audit_history/tests/test_audit_history.py (hunk 1 above)
Comment (hheimbuerger) [reply audit_history/tests/test_audit_history.py]: Ah yes, makes sense! Forgot about that.

File: audit_history/tests/test_audit_history.py (hunk 2 above)
Comment (Killsoft1692) [reply audit_history/tests/test_audit_history.py]: Updated. Please check https://github.com/nexto/django-model-audit-history/pull/4/commits/bf318757c7569e1099e824eb85679c48df4b33c0

File: audit_history/mixins.py (hunk 3 above)
Comment (Killsoft1692) [reply audit_history/mixins.py:45]: Payload can have any data that we can imagine. So in my case we ensure that when we save this data we don't receive TypeError: datetime.datetime(2019, 1, 30, 11, 56, 33, 369765, tzinfo=<UTC>) is not JSON serializable orTypeError when we have Decimal in payload

File: audit_history/utils.py (hunk 4 above)
Comment (Killsoft1692) [reply audit_history/utils.py:5]: Yep. We put data and if this data can be serialized we return it if not we convert it to serializable data and return it
